cyipopt/scipy_interface.py

```diff
@@ -691,3 +714,13 @@ def _minimize_ipopt_iv(fun, x0, args, kwargs, method, jac, hess, hessp,
 
     return (fun, x0, args, kwargs, method, jac, hess, hessp,
             bounds, constraints, tol, callback, options)
+
+def _dual_initial_guess_validation(dual_name: str, dual_value: list, length: int):
+    if not isinstance(dual_value, list):
+        raise TypeError(f'`{dual_name}` must be a list.')
```

Comment:
Why does this have to be a list? Couldn't any iterable or sequence work?

Reply:
Since `Problem.solve()` uses empty lists as the default values for `lagrange`, `zl` and `zu`, and create zero numpy arrays when they are `[]`, I just followed it. Indeed any iterable or sequence would work.
https://github.com/mechmotum/cyipopt/blob/20488262bae5647f6af0c176007631cbc687ec9d/cyipopt/cython/ipopt_wrapper.pyx#L628